lib/rucio/common/utils.py

```diff
@@ -63,8 +57,9 @@
         EXTRA_MODULES['paramiko'] = False
 
 if TYPE_CHECKING:
-    from collections.abc import Callable, Iterable, Iterator, Sequence
+    from collections.abc import Callable, Iterable, Iterator, Mapping, Sequence
     T = TypeVar('T')
+    Q = TypeVar('Q')
```

Comment:
T and Q are not very descriptive type names. Is this a common convention I don't know about? Should they be renamed to something more verbose?

Reply:
That's the common convention I see around:
- https://docs.python.org/3/library/typing.html#typing.TypeVar
- https://stackoverflow.com/questions/74589610/whats-pylints-typevar-name-specification

Which seems to be close to the PEP 8 specification: https://peps.python.org/pep-0008/#type-variable-names

In the context of these typevars being used in `invert_dict`, I agree that they could be more explicit, I can rename them to have `Hashable` in the name (since they both need to be hashable for `invert_dict`) to work.
`T` is also used in `retrying`, where I'd leave it as `T` considering that it's more generic.

Reply:
Addressed in latest revision